ccm_web/client/src/pages/Home.tsx

```diff
@@ -33,7 +36,38 @@ interface HomeProps extends LtiProps {
 }
 
 function Home (props: HomeProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
   const classes = useStyles()
+  const [course, setCourse] = useState<undefined|CanvasCourseBase>(undefined)
+
+  const [doLoadCourse, isLoadingCourse, getCourseError] = usePromise(
+    async () => await getCourse(props.ltiKey, props.globals.course.id),
+    (value: CanvasCourseBase) => setCourse(value)
+  )
+
+  useEffect(() => {
+    void doLoadCourse()
+  }, [])
+
+  const setCourseNameAsync = async (
+    newCourseName: string
+  ): Promise<CanvasCourseBase|undefined> => {
+    return await apiSetCourseName(props.ltiKey, props.globals.course.id, newCourseName)
+  }
+
+  const [doSetCourseName, setCourseNameLoading, setCourseNameError] = usePromise<CanvasCourseBase|undefined, typeof setCourseNameAsync>(
```

Comment:
`setCourseNameLoading` is not being used and linting error is shown? can be be used in the code?

Reply:
It can, i don't imagine it will provide much functionality though :-p

Reply:
Something we did in ROHQ was to have an aggregate `isLoading` variable that combined all the loading booleans like `setCourseNameLoading | getCourseLoading` and then we'd use that to determine whether to show a spinner in the bottom right hand corner. From a UX perspective, it's nice to know the site is doing some work. The problem of when it happens too quickly (which @chrisrrowland referenced elsewhere) is tricky, but I think this aggregate approach is sound.

Reply:
This is no longer relevant after refactoring